### DISCUSSION TITLE
Prevent private attributes from showing up in JSON response

### DISCUSSION THREAD

**Discussion by aceenprsu** [fastapi/fastapi#10378] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic.dataclasses import dataclass
from uuid import uuid4

@dataclass
class Example:
    _uid: str
    value: str

app = FastAPI()

@app.get('/example') 
async def endpoint() -> Example:
    return Example(value="example", _uid=str(uuid4()))
```

### Description

* Open the browser and call the endpoint `/example`
* It returns a JSON with `{"_uid":"87bea7a8-3515-4065-a2d7-4cbb5db7dc72","value":"example"}`
* But I expected it to return `{"value":"example"}`

How can I prevent the private attributes from the Example dataclass from showing up in the endpoint response?

With an older FastAPI version (before pydantic 2 support) the response did not include the private fields from the pydantic model.

### Operating System

Linux

### FastAPI Version

0.103.2

### Pydantic Version

2.4.2

### Python Version

3.10.6

**Comment by Kludex** [comment]
I see a warning on previous versions of FastAPI when using `pydantic.dataclasses` or `dataclasses` itself:

### Pydantic Dataclasses
```bash
/Users/marcelotryle/dev/pydantic/main.py:7: RuntimeWarning: fields may not start with an underscore, ignoring "_uid"
  @dataclass
```

### Standard Library Dataclasses
```bash
/Users/marcelotryle/.pyenv/versions/3.11.1/envs/fastapi-tmp/lib/python3.11/site-packages/fastapi/utils.py:99: RuntimeWarning: fields may not start with an underscore, ignoring "_uid"
  return ModelField(**kwargs)  # type: ignore[arg-type]
```

When using `BaseModel`, the field prefixed underscored fields are present in the response.
